### DISCUSSION TITLE
how to print the training set accuracy in the json file

### DISCUSSION THREAD

**Discussion by ghty11468** [open-mmlab/mmclassification#1274] [Q&A]
In the training process, there is only the validation set accuracy in the json file, how to print the training set accuracy in the json file

**Comment by Ezra-Yu** [comment]
set the ``model.head.cal_acc`` to  True, refer to https://github.com/open-mmlab/mmclassification/blob/46af7d3ed2538dea3133c06250dd3683fd792e38/mmcls/models/heads/cls_head.py#L22

The following is an example:
```
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='ResNeSt',
        depth=50,
        num_stages=4,
        out_indices=(3, ),
        style='pytorch'),
    neck=dict(type='GlobalAveragePooling'),
    head=dict(
        type='LinearClsHead',
        num_classes=1000,
        in_channels=2048,
        loss=dict(
            type='LabelSmoothLoss',
            label_smooth_val=0.1,
            num_classes=1000,
            reduction='mean',
            loss_weight=1.0),
        topk=(1, 5),
        cal_acc=True))
```

**Reply by ghty11468** [reply]
Thanks for your answer, but this method seems to print the accuracy for each iteration. Is there any other method that prints the accuracy of each epoch?

**Reply by Ezra-Yu** [reply]
Maybe you can let your train dataset be the same as val dataset.

**Reply by ghty11468** [reply]
This requires two training sessions, and the two sessions do not correspond to each other. Is there any way to print both training dataset and validation dataset accuracy of epoch by training once.To get the graph in the paper. Looking forward to your reply

**Reply by Ezra-Yu** [reply]
set `cal_acc=True` and then write a [hook](https://mmengine.readthedocs.io/en/latest/tutorials/hook.html) to collect all the iters and ACC for each iter [(iter, ACC), (iter, ACC) ....], then calculate the overall accuracy of the training epoch.

Or just inheriting and rewriting the [LoggerHook](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/logger_hook.py).

**Reply by ghty11468** [reply]
This method is difficult for myself. I will look for other methods afterwards. Thank you very much for your reply!
